databroker/broker.py

```diff
@@ -924,8 +925,10 @@ def wrapped_f(*args, **kwargs):
                 if name == 'descriptor':
                     # Mutate fs_doc here to mark data as external.
                     for data_name in external_writers.keys():
-                        fs_doc['data_keys'][data_name].update(
-                            external='FILESTORE:')
+                        # data doesn't have to exist?
+                        if data_name in fs_doc['data_keys']:
```

Comment:
As we discussed on slack: it's OK if the `store_dec` includes some keys that aren't used sometimes. The decorator underneath (i.e., `event_map`) is the one that specifies the exact keys for a certain operation.